libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1013,35 +1023,41 @@ void IRGeneratorForStatements::endVisit(Identifier const& _identifier)
 			else
 			{
 				solAssert(_identifier.name() == "this", "");
-				defineExpression(_identifier) << "address()\n";
+				define(_identifier) << "address()\n";
 			}
 			break;
 		case Type::Category::Integer:
 			solAssert(_identifier.name() == "now", "");
-			defineExpression(_identifier) << "timestamp()\n";
+			define(_identifier) << "timestamp()\n";
 			break;
 		default:
 			break;
 		}
 		return;
 	}
 	else if (FunctionDefinition const* functionDef = dynamic_cast<FunctionDefinition const*>(declaration))
-		defineExpression(_identifier) << to_string(m_context.virtualFunction(*functionDef).id()) << "\n";
+		define(_identifier) << m_context.virtualFunction(*functionDef).id() << "\n";
```

Comment:
I would really like to avoid string stream's implicit integer to string conversion. It depends on the state of the stream and does magic things.